packages/contracts-bedrock/test/Specs.t.sol

```diff
@@ -894,4 +897,9 @@ contract Specification_Test is CommonTest {
         }
         assertEq(numSpecs, numCheckedEntries, "Some specs were not checked");
     }
+
+    /// @notice Ensures that the DeputyGuardian is authorized to take all Guardian actions.
+    function testDeputyGuardianAuth() public {
+        assertEq(specsByRole[Role.DEPUTYGUARDIAN].length, specsByRole[Role.GUARDIAN].length);
```

Comment:
Any ideas on how we can check the set of functions is identical? For example if `blacklistDisputeGame` is replaced with `foo` in the portal, but we don't change to `foo` in the DGM, this test still passes. We can't use selectors exactly unfortunately because of the `pause()` vs. `pause(string)` difference

Reply:
I am not surprised by this question. :)

Since the arguments differ on all functions, we'd need either: 
1. a refactor to track the `name` on within the `Spec` struct alongside the selector.
2. a brute force approach making 3 assertions per DGM method (taking `blacklistDisputeGame` as an example): 
	a. The DGM abi has a `blacklistDisputeGame` method.
	b. The `DGM.blacklistDisputeGame` spec has `spec.auth == Role.DEPUTYGUARDIAN`. 
	c. The `OptimismPortal2` abi has a `blacklistDisputeGame` method.
	d. `OptimismPortal2.blacklistDisputeGame` spec has `spec.auth == Role.GUARDIAN`. 
	
Do you see a reason that the refactor would have benefits in the future? If not I'd lean towards the second.

Reply:
I'm not sure I fully follow the brute force approach, but the set of methods is small enough that I'm onboard with that. 

I think what you are saying is just compare method names and assert that each `{ProtocolContract}.{methodName}. GUARDIAN ` has a corresponding `DGM.{identicalMethodName-notSelector}.DEPUTYGUARDIAN `? Either way, feel free to respond with just the implementation code if that's easier

Reply:
PTAL at the diff now.